Add Azure attestation functionality

longtomjr opened the pull request.

longtomjr committed:
feat(uenclave): add attestation types + client

longtomjr committed:
feat(uenclave): refactor and impl attestation call

longtomjr committed:
feat(data_service): update actor to use new method

longtomjr committed:
feat(data_service): update config and config usage

As part of the config update, I made the
EnclaveActor use a shared config instead
of copying a config value.

PiDelport commented:
(I rebased for the Makefile conflicts, and amended the last commit to sync `Cargo.lock`.)

PiDelport committed:
refactor(rtc_uenclave): uncomment sgx_dcap_ql link, for now

This simplifies building and testing in the interim.

PiDelport committed:
fix(rtc_uenclave): only derive Debug on RtcEnclave when not testing

When testing, the mock values don't currently implement Debug, which
makes the derive fail.

PiDelport committed:
docs(rtc_uenclave): document intermittent dcap_azure_attestation_works failure

PiDelport committed:
docs(README): add "Running tests", with note about test thread concurrency

PiDelport reviewed (approved):
Alright, the tests run locally. 🎉
